dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/filters/ApiAuthorizationFilter.java

```diff
@@ -0,0 +1,77 @@
+package edu.harvard.iq.dataverse.api.filters;
+
+import edu.harvard.iq.dataverse.DataverseSession;
+import edu.harvard.iq.dataverse.UserServiceBean;
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUser;
+import edu.harvard.iq.dataverse.persistence.user.GuestUser;
+
+import javax.inject.Inject;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+public class ApiAuthorizationFilter implements Filter {
+
+    private DataverseSession session;
+    private AuthenticationServiceBean authenticationService;
+    private UserServiceBean userService;
+
+    // -------------------- CONSTRUCTORS ---------------------
+
+    @Inject
+    public ApiAuthorizationFilter(DataverseSession session, AuthenticationServiceBean authenticationService, UserServiceBean userService) {
+        this.session = session;
+        this.authenticationService = authenticationService;
+        this.userService = userService;
+    }
+
+    // -------------------- LOGIC ---------------------
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        boolean loggedByToken = logUserByTokenIfNeeded(servletRequest);
+        chain.doFilter(servletRequest, response);
+        logoutIfLoggedByToken(servletRequest, loggedByToken);
```

Comment:
Imo extracting "if (loggedByToken)" here would be little cleaner, since atm method can be called but there would be no action taken inside. And if you extract it here, the method would only be executed if action is needed. For me it's little more logical.